apps/dotcom/client/src/tla/app/TldrawApp.ts

```diff
@@ -219,123 +223,246 @@ export class TldrawApp {
-		delete newFilePartial.lastPublished
-		delete newFilePartial.publishedSlug
-		delete newFilePartial.published
+		if (!file.published) return Result.ok('success')
 
-		const newFile = TldrawAppFileRecordType.create(newFilePartial as TldrawAppFile)
+		// Optimistic update
+		this.store.put([{ ...file, published: false }])
 
-		const editorStoreSnapshot = getCurrentEditor()?.store.getStoreSnapshot()
-		this.store.put([newFile])
+		const fileSlug = fileId.split(':')[1]
 
-		return { newFile, editorStoreSnapshot }
-	}
+		const res = await fetch(`${PUBLISH_ENDPOINT}/${fileSlug}`, {
+			method: 'DELETE',
+			headers: {
+				'Content-Type': 'application/json',
+				Authorization: `Bearer ${token}`,
+			},
+		})
 
-	isFileOwner(fileId: TldrawAppFileId) {
-		const file = this.get(fileId) as TldrawAppFile
-		if (!file) return false
-		return file.ownerId === this.getCurrentUserId()
-	}
+		const response = (await res.json()) as UnpublishFileResponseBody
 
-	async deleteOrForgetFile(fileId: TldrawAppFileId) {
-		if (this.isFileOwner(fileId)) {
-			this.store.remove([
-				fileId,
-				...this.getAll('file-state')
-					.filter((r) => r.fileId === fileId)
-					.map((r) => r.id),
-			])
-		} else {
-			const ownerId = this.getCurrentUserId()
-			const fileStates = this.getAll('file-state')
-				.filter((r) => r.fileId === fileId && r.ownerId === ownerId)
-				.map((r) => r.id)
-			this.store.remove(fileStates)
+		if (!res.ok || response.error) {
+			// Revert optimistic update
+			this.store.put([{ ...file, published: true }])
+			return Result.err('could not unpublish')
 		}
-	}
 
-	async createFilesFromTldrFiles(_snapshots: TLStoreSnapshot[]) {
-		// todo: upload the files to the server and create files locally
-		console.warn('tldraw file uploads are not implemented yet, but you are in the right place')
-		return new Promise((r) => setTimeout(r, 2000))
+		return Result.ok('success')
 	}
 
-	async createSnapshotLink(editor: Editor, parentSlug: string, fileSlug: string, token: string) {
-		const data = await getSnapshotData(editor)
+	/**
+	 * Remove a user's file states for a file and delete the file if the user is the owner of the file.
+	 *
+	 * @param fileId - The file id.
+	 * @param token - The user's token.
+	 */
+	async deleteOrForgetFile(fileId: TldrawAppFileId, token: string) {
+		// Stash these so that we can restore them later
+		let fileStates: TldrawAppFileState[]
+		const file = this.get(fileId) as TldrawAppFile
 
-		if (!data) return Result.err('could not get snapshot data')
+		if (this.isFileOwner(fileId)) {
+			// Optimistic update, remove file and file states
+			const userId = this.getCurrentUserId()
+			fileStates = this.getAll('file-state').filter(
+				(r) => r.fileId === fileId && r.ownerId === userId
+			)
+			this.store.remove([fileId, ...fileStates.map((s) => s.id)])
+		} else {
+			// If not the owner, just remove the file state
+			fileStates = this.getAll('file-state').filter((r) => r.fileId === fileId)
```

Comment:
```suggestion
			fileStates = this.getAll('file-state').filter(
				(r) => r.fileId === fileId
			)
			this.store.remove([fileId, ...fileStates.map((s) => s.id)])
		} else {
			// If not the owner, just remove the file state
			fileStates = this.getAll('file-state').filter((r) => r.fileId === fileId && r.ownerId === userId)
```

you mistakenly switched these predicates around